### ISSUE TITLE
Allow the visibility of the remix button to be "toggle-able"

### ISSUE THREAD

**Issue by gideonthomas** [mozilla/thimble.mozilla.org#1515]
Moved from a discussion in https://github.com/mozilla/thimble.mozilla.org/issues/1503. We should allow the remix button to be hidden if the user prefers, since it can interfere with the user's project design.

Needs: 
- Designs on Thimble to "Show/Hide Remix button" (the project is still remixable, just not through the UI).
- Backend implementation in publish.wm.org to decide whether to inject the script or not.

cc @flukeout

**Comment by gideonthomas** [comment]
In response to https://github.com/mozilla/thimble.mozilla.org/issues/1503#issuecomment-233375557 :

Yeah it will need to be tracked in the db.

Hmm, yeah true. I don't know if it is super important that we market the Thimble brand in projects. On the other hand, one way we could approach this is to have the "Hide" button inside the remix bar as well so that if the creator of the project allows the remix bar to be shown, the viewer of the project can decide whether they want to see the remix bar or not as well. That way we give some control to both the creator as well as the viewer.

@hannahkane do you have any thoughts on this?

**Comment by flukeout** [comment]
To tack on another need:

* When the "Show Remix" setting is toggled on a published project, the "Update Published Project" button should appear so the user can make the change take effect.

**Comment by gideonthomas** [comment]
@flukeout we could, or it can happen automatically! Whatever you think is the best UX here.

**Comment by samueller** [comment]
I like the hide button to give some control to the user too. The "show remix" setting updating the published project automatically makes sense since that setting's only purpose is affecting the published project.

**Comment by hannahkane** [comment]
+1 to @samueller's reasoning on the points above.

**Comment by flukeout** [comment]
Agreed on updating the setting automatically on the Published project.

Not convinced about offering a way for someone who is viewing a project to hide the Remix button, however. If the main problem with the button is that it is in the way of a particular design, it shouldn't be there to begin with.

**Comment by hannahkane** [comment]
@flukeout - are you suggesting a different solution altogether then? Are you saying that even if the author hasn't chosen to toggle the Remix button off, there are circumstances where we would remove it anyway?

**Comment by flukeout** [comment]
Oh no. There was a separate suggestion to also allow the person viewing a project to hide the Remix button if they wanted to when they were viewing it. Not related to the toggle that the author would get.

**Comment by hannahkane** [comment]
Yeah, I get that. Sorry, this is confusing. I didn't understand your [comment](https://github.com/mozilla/thimble.mozilla.org/issues/1515#issuecomment-233399737) about "If the problem is that it's in the way, it shouldn't be there to begin with." 

My understanding of @gideonthomas's suggestion above was that the Hide button would be available to the viewer in the case where the author had chosen *not* to disable the Remix button. Your comment made me think you were suggesting that the Remix button should somehow *never* appear if it was blocking any content.

Can you rephrase what you were suggesting? Thanks!

**Comment by flukeout** [comment]
Sorry for muddying the waters here. The following bullets summarize my thoughts

* Allow the author to disable the Remix button
* Do NOT allow the viewer to hide the Remix button if it is present

Rationale
* If the Remix button conflicts with the design, the author would have disabled it anyway

**Comment by samueller** [comment]
+1 for the summary by @flukeout

**Comment by hannahkane** [comment]
+1. I can imagine use cases for giving control to the viewer, but probably best to not over-complicate now.

**Comment by flukeout** [comment]
@hannahkane agreed on possible use cases. We can address in a subsequent effort if we think it's necessary. For now, here's what I was thinking for the Publish dialog...

![image](https://cloud.githubusercontent.com/assets/25212/16968101/a6a9ea32-4dc1-11e6-8262-7c16353fd260.png)

**Comment by samueller** [comment]
@flukeout that looks great

**Comment by flukeout** [comment]
Haha, thanks! @gideonthomas I can add this markup and leave you to wire it up?

@samueller Do you think we need any additional context around that toggle? Maybe a line underneath that says...

``Remixing allows others to make their own version of your project``

Or something to that effect?

**Comment by samueller** [comment]
@flukeout, I subscribe to the philosophy of less is more. I think you have the right phrase, it should be clear enough for anybody publishing their website.

**Comment by gideonthomas** [comment]
@flukeout looks great! Yep I can hook it up if you add the markup

**Comment by khanh2003** [comment]
Why not also movable?
================

**Comment by flukeout** [comment]
@gideonthomas Okay, I'll add the markup in the heartbeat and keep you posted.

@khanh2003 We can investigate the need and solutions for a movable Remix button in a subsequent issue, but will keep the work in this issue scoped to the 'toggable' feature.

Do you have any suggestions or ideas about how to make the button movable? Who would control where the button goes, the author or the viewer? Would the position be remembered for subsequent projects or just for that one? What specific problem does a movable button solve? Curious about your thoughts - cheers.

**Comment by khanh2003** [comment]
The viewer would control where the button goes. The position is remembered.
=
Specific problem a movable button solve:
=
```
LOOK AT THIS:
https://thimbleprojects.org/ucode/87099/
We should put the button in a page, inside the page there is an iframe that shows the user's project.
```

**Comment by samueller** [comment]
Curious what the status of this toggle-able remix button is. We are teaching using Thimble now and this is an issue.

**Comment by gideonthomas** [comment]
@samueller unfortunately, it doesn't look like we are able to get to this issue in this iteration. We are however working on figuring out the right UI/UX for this.

For the time-being, might I offer you a temporary quick fix? Try adding this small bit of code to your HTML file(s), and it should work:
```html
<style>
  .details-bar {
    display: none !important;
  }
</style>
```

**Comment by flukeout** [comment]
Closing, refiled in #1623
